lib/identity_doc_auth/error_generator.rb

```diff
@@ -0,0 +1,195 @@
+      front_glare_fail, back_glare_fail = false
+
+      processed_image_metrics.each do |side, img_metrics|
+        hdpi = img_metrics['HorizontalResolution']&.to_i || 0
+        vdpi = img_metrics['VerticalResolution']&.to_i || 0
+        if hdpi < dpi_threshold || vdpi < dpi_threshold
+          front_dpi_fail = true if side == :front
+          back_dpi_fail = true if side == :back
+        end
+
+        sharpness = img_metrics['SharpnessMetric']&.to_i
+        if sharpness.present? && sharpness < sharpness_threshold
+          front_sharp_fail = true if side == :front
+          back_sharp_fail = true if side == :back
+        end
+
+        glare = img_metrics['GlareMetric']&.to_i
+        if glare.present? && glare < glare_threshold
+          front_glare_fail = true if side == :front
+          back_glare_fail = true if side == :back
+        end
+      end
+
+      return { GENERAL => [Errors::DPI_LOW_BOTH_SIDES] } if front_dpi_fail && back_dpi_fail
+      return { GENERAL => [Errors::DPI_LOW_ONE_SIDE] } if front_dpi_fail || back_dpi_fail
+
+      return { GENERAL => [Errors::SHARP_LOW_BOTH_SIDES] } if front_sharp_fail && back_sharp_fail
+      return { GENERAL => [Errors::SHARP_LOW_ONE_SIDE] } if front_sharp_fail || back_sharp_fail
+
+      return { GENERAL => [Errors::GLARE_LOW_BOTH_SIDES] } if front_glare_fail && back_glare_fail
+      return { GENERAL => [Errors::GLARE_LOW_ONE_SIDE] } if front_glare_fail || back_glare_fail
+
+      {}
+    end
+
+    def get_error_messages(liveness_enabled, response_info)
+      errors = Hash.new { |hash, key| hash[key] = Set.new }
+
+      if response_info[:doc_auth_result] != 'Passed'
+        response_info[:processed_alerts][:failed]&.each do |alert|
+          alert_msg_hash = ALERT_MESSAGES[alert[:name].to_sym]
+
+          if alert_msg_hash.present?
+            errors[alert_msg_hash[:type]] << alert_msg_hash[:msg_key]
+          end
+        end
+      end
+
+      pm_results = response_info[:PortraitMatchResults] || {}
+      if liveness_enabled && pm_results.dig(:FaceMatchResult) != 'Pass'
+        errors[SELFIE] << Errors::SELFIE_FAILURE
+      end
+
+      errors
+    end
+
+    def general_error(liveness_enabled)
+      if liveness_enabled
+        Errors::GENERAL_ERROR_LIVENESS
+      else
+        Errors::GENERAL_ERROR_NO_LIVENESS
+      end
+    end
+
+    def scan_for_unknown_alerts(response_info)
+      all_alerts = [*response_info[:processed_alerts][:failed], *response_info[:processed_alerts][:passed]]
+      unknown_fail_count = 0
+
+      unknown_alerts = []
+      all_alerts.each do |alert|
+        if ALERT_MESSAGES[alert[:name].to_sym].blank?
+          unknown_alerts.push(alert[:name])
+
+          if alert[:result] != 'Passed'
+            unknown_fail_count += 1
+          end
+        end
+      end
+
+      return 0 if unknown_alerts.empty?
+
+      message = 'LN TrueID responded with alert name(s) we do not handle: ' + unknown_alerts.to_s
```

Comment:
this is now a shared error generator, should not have specific vendors in the name right?

Reply:
Oops, yeah. I forgot about those messages. Fixed them locally and will push shortly.